packages/sdk/contractkit/src/wrappers/Escrow.test.ts

```diff
@@ -0,0 +1,126 @@
+  let federatedAttestations: FederatedAttestationsWrapper
+  let stableTokenContract: StableTokenWrapper
+  let identifier: string
+
+  beforeAll(async () => {
+    escrow = await kit.contracts.getEscrow()
+    stableTokenContract = await kit.contracts.getStableToken()
+    federatedAttestations = await kit.contracts.getFederatedAttestations()
+
+    accounts = await web3.eth.getAccounts()
+    kit.defaultAccount = accounts[0]
+
+    identifier = kit.web3.utils.soliditySha3({
+      t: 'bytes32',
+      v: kit.web3.eth.accounts.create().address,
+    }) as string
+  })
+
+  it('transfer with trusted issuers should set TrustedIssuersPerPayment', async () => {
+    const testPaymentId = kit.web3.eth.accounts.create().address
+    await federatedAttestations
+      .registerAttestationAsIssuer(identifier, kit.defaultAccount as string, TIMESTAMP)
+      .sendAndWaitForReceipt()
+
+    await stableTokenContract.approve(escrow.address, TEN_CUSD).sendAndWaitForReceipt()
+
+    await escrow
+      .transferWithTrustedIssuers(
+        identifier,
+        stableTokenContract.address,
+        TEN_CUSD,
+        1000,
+        testPaymentId,
+        1,
+        accounts
+      )
+      .sendAndWaitForReceipt()
+
+    const trustedIssuersPerPayment = await escrow.getTrustedIssuersPerPayment(testPaymentId)
+
+    expect(trustedIssuersPerPayment[0]).toEqual(kit.defaultAccount)
+  })
+  it('withdraw should be successful after transferWithTrustedIssuers', async () => {
+    const sender: string = accounts[1]
+    const receiver: string = accounts[2]
+    const withdrawKeyAddress: string = accounts[3]
+    const oneDayInSecs: number = 86400
+    const parsedSig = await getParsedSignatureOfAddress(web3, receiver, withdrawKeyAddress)
+
+    await federatedAttestations
+      .registerAttestationAsIssuer(identifier, receiver, TIMESTAMP)
+      .sendAndWaitForReceipt()
+
+    const senderBalanceBefore = await stableTokenContract.balanceOf(sender)
+    const receiverBalanceBefore = await stableTokenContract.balanceOf(receiver)
+
+    await stableTokenContract
+      .approve(escrow.address, TEN_CUSD)
+      .sendAndWaitForReceipt({ from: sender })
+
+    await escrow
+      .transferWithTrustedIssuers(
+        identifier,
+        stableTokenContract.address,
+        TEN_CUSD,
+        oneDayInSecs,
+        withdrawKeyAddress,
+        1,
+        accounts
+      )
+      .send({ from: sender })
+
+    await escrow
+      .withdraw(withdrawKeyAddress, parsedSig.v, parsedSig.r, parsedSig.s)
+      .sendAndWaitForReceipt({ from: receiver })
+
+    const senderBalanceAfter = await stableTokenContract.balanceOf(sender)
+    const receiverBalanceAfter = await stableTokenContract.balanceOf(receiver)
+
+    expect(senderBalanceBefore.minus(+TEN_CUSD)).toEqual(senderBalanceAfter)
+    expect(receiverBalanceBefore.plus(+TEN_CUSD)).toEqual(receiverBalanceAfter)
+  })
+  it('withdraw should revert if attestation is not registered', async () => {
```

Comment:
would also be great to do one where withdraw fails when attestation is registered by an issuer not on the trusted issuer list :)

Reply:
Ah yes def agree that this is a good test although, I do think it doesn't fit into the scope of this suite. While we do test different flows, the aim here is to test the wrapper functionality as opposed to alternate flows/failure paths

Reply:
I see both sides here. I agree that this case probably should be tested at the protocol level, but it looks like it's also missing there. @Jesse-Sawa maybe you could just put a TODO(ASv2) statement on this line to help us remember to add it? https://github.com/celo-org/celo-monorepo/blob/1522250ff90d0ab1d8afd328d32ac15ce160b1e5/packages/protocol/test/identity/escrow.ts#L1121

Reply:
Sure thing - I'll actually add it to the "Add registerAttestation test" issue and knock them both out in a PR